pkg/plugin/vector/store/qdrant.go

```diff
@@ -82,14 +83,89 @@ func (q *qdrantStore) CollectionExists(ctx context.Context, collection string) (
 	return true, nil
 }
 
-func (q *qdrantStore) Search(ctx context.Context, collection string, vector []float32, topK uint64) ([]SearchResult, error) {
+func (q *qdrantStore) mapFilters(ctx context.Context, filter map[string]interface{}) (*qdrant.Filter, error) {
+	qdrantFilterMap := &qdrant.Filter{}
+
+	if filter == nil {
+		return qdrantFilterMap, nil
+	}
+
+	for k, v := range filter {
+		switch v := v.(type) {
+		case map[string]interface{}:
+			for op, val := range v {
+				match, err := createQdrantMatch(val)
+				if err != nil {
+					return nil, err
+				}
+
+				condition := &qdrant.Condition{
+					ConditionOneOf: &qdrant.Condition_Field{
+						Field: &qdrant.FieldCondition{
+							Key:   k,
+							Match: match,
+						},
+					},
+				}
+
+				switch op {
+				case "$eq":
+					qdrantFilterMap.Must = append(qdrantFilterMap.Must, condition)
+				case "$ne":
+					qdrantFilterMap.MustNot = append(qdrantFilterMap.MustNot, condition)
+				}
```

Comment:
We should probably bail out if neither of these match, same for the `$or` / `$and` in the other case.